vm/parser.ts

```diff
@@ -933,16 +1153,22 @@ module J2ME {
     }
     
     getAllInterfaces(): ClassInfo [] {
+      if (this.allInterfaces) {
+        return this.allInterfaces;
+      }
```

Comment:
It isn't clear how often this is called, and it does cache the result, but *ArrayUtilities.pushMany* and *unique* are both pretty inefficient, so we should take a closer look at the cost of these changes.